### PR TITLE
remove thriftpy2 ceiling

### PR DESCRIPTION
**Pull request by jameslamb**
#4496 introduced a ceiling on `thriftpy2`. Context: https://github.com/rapidsai/cugraph/pull/4496#issuecomment-2192783353

The bug that ceiling was added to avoid was fixed in v0.5.2 of `thriftpy2`, which was just released (https://github.com/rapidsai/cugraph/pull/4478#issuecomment-2210762587).

This removes that, adding `!=` constraints to skip the 2 versions that `cugraph` was not compatible with.

### Why not a floor?

I'm proposing adding `!=` constraints to skip v0.5.0 and v0.5.1 to maximize `cugraph`'s compatibility with other environments... that'd allow it to be used in environments with `thriftpy2<0.5.0` and in environments with `thriftpy2>0.5.2`.

Let me know if you'd prefer the simplicity of a floor like `>=0.5.2` instead.

### CODE REVIEW

**Comment by jameslamb** [comment]
This is blocked until #4522 is merged.

**Review by bdice — APPROVED** [review]

**Review by nv-rliu — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]

**Pull request merged**
